### PR TITLE
Implement basic Appointment and AppointmentList

### PR DESCRIPTION
**Pull request by tanweili**
AppointmentList will be the main interface to store appointment
related information. Features added are adding new appointment,
showing all appointment and finding by patient name, doctor name,
nric or date. More code still needs to be added to Parse and
Command to accomodate this addition.

### CODE REVIEW

**Comment by tanweili** [comment]
To facilitate proper add appointment feature, will require find patient and doctor nric feature to check if input is valid. @wraineflores

File: src/main/java/seedu/duke/assets/Appointment.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package seedu.duke.assets;
+
+public class Appointment {
```
Comment (wraineflores) [inline src/main/java/seedu/duke/assets/Appointment.java:3]: Looks good!

File: src/main/java/seedu/duke/assets/AppointmentList.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        appointments.add(newAppointment);
+    }
+
+    public void showAll() {
+        for (int i = 0; i < appointments.size(); i++) {
+            appointments.get(i).show();
+        }
+    }
+
+    public void find(String criteria, String input) {
```
Comment (shxr3f) [inline src/main/java/seedu/duke/assets/AppointmentList.java:25]: Great way of implementing the find method!

**Review by shxr3f — APPROVED** [review]
Everything looks fine to me!

**Review by kktai1512 — APPROVED** [review]
lgtm!

**Comment by shxr3f** [comment]
Do update your print statements to use the print methods given in the UI Class.

File: src/main/java/seedu/duke/helper/Parser.java [hunk 3]
```diff
@@ -68,6 +55,32 @@ private static boolean validateAddDoctor(String[] parameters) {
...
+        } else {
+            return null;
+        }
+    }
+
+    private static boolean validateAddDoctor(String[] parameters) {
+        boolean isValid = validateAddPerson(Arrays.copyOfRange(parameters, 0, 6));
+        //validate full name cause specialization is also j a name
+        if (!validateFullName(parameters[6])) {
+            System.out.println("Specialization must be a name");
```
Comment (shxr3f) [inline src/main/java/seedu/duke/helper/Parser.java:78]: Missing UI print method

File: src/main/java/seedu/duke/assets/Appointment.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        this.patientName = patientName;
+        this.patientNric = patientNric;
+        this.doctorName = doctorName;
+        this.doctorNric = doctorNric;
+        this.appointmentDate = appointmentDate;
+        this.appointmentDetails = appointmentDetails;
+    }
+
+    public void show() {
+        System.out.println("-------------------------");
```
Comment (shxr3f) [inline src/main/java/seedu/duke/assets/Appointment.java:22]: Should we be using printParagraph here>

**Comment by tanweili** [comment]
@kktai1512 Abstract validator probably needs to account for new code added for Appointment when merging. @wraineflores will implement finding after your filter/finding for doctor/patient

File: src/main/java/seedu/duke/assets/AppointmentList.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            break;
+        case "nric":
+            for (int i = 0; i < appointments.size(); i++) {
+                if (appointments.get(i).getPatientNric().equals(input)) {
+                    appointments.get(i).show();
+                }
+            }
+            break;
+        default:
+            System.out.println("Invalid search criteria! The valid criteria are:");
```
Comment (shxr3f) [inline src/main/java/seedu/duke/assets/AppointmentList.java:56]: Similar here.

File: src/main/java/seedu/duke/helper/Parser.java [hunk 6]
```diff
@@ -102,14 +115,46 @@ private static boolean validateAddPerson(String[] parameters) {
 
     private static boolean validateAddPatient(String[] parameters) {
         boolean isValid = validateAddPerson(Arrays.copyOfRange(parameters, 0, 6));
-        if (!validateAdmissionDate(parameters[6])) {
-            UI.printParagraph("Date of birth must be in YYYY-MM-DD format. "
+        if (!validateDate(parameters[6], "patient")) {
+            System.out.println("Date of birth must be in YYYY-MM-DD format. "
```
Comment (shxr3f) [inline src/main/java/seedu/duke/helper/Parser.java:119]: Need to change print statement here.

File: src/main/java/seedu/duke/helper/Parser.java [hunk 7]
```diff
@@ -102,14 +115,46 @@ private static boolean validateAddPerson(String[] parameters) {
...
                     + "It cannot be before 1980-01-01 or be today and after.");
             isValid = false;
         }
         return isValid;
     }
 
+    private static boolean validateAddAppointment(String[] parameters) {
```
Comment (shxr3f) [inline src/main/java/seedu/duke/helper/Parser.java:126]: the print statements in this method arent using the UI method either

File: src/main/java/seedu/duke/helper/Storage.java [hunk 8]
```diff
@@ -111,6 +126,27 @@ private void savePatientData() {
         }
     }
 
+    private void saveAppointmentData() {
```
Comment (shxr3f) [inline src/main/java/seedu/duke/helper/Storage.java:129]: Here as well!

**Review by shxr3f — COMMENTED** [review]
Some places do not use the ui print statements, are these intended?

**Pull request merged**